host/sgx/create.c

```diff
@@ -323,12 +324,74 @@ static oe_result_t _add_control_pages(
     return result;
 }
 
+#define OE_LOAD_EXTRA_ENCLAVE_DATA_HOOK_ARG_MAGIC 0x793d33e0efb446d0
+
+typedef struct oe_load_extra_enclave_data_hook_arg
+{
+    uint64_t magic;
+    oe_sgx_load_context_t* sgx_load_context;
+    uint64_t enclave_base;
+    uint64_t enclave_start;
+    uint64_t base_vaddr; /* address relative to the enclave start */
+    uint64_t vaddr;      /* address relative to the extra data start */
+} oe_load_extra_enclave_data_hook_arg_t;
+
+/* This weak form may be overriden by the enclave application */
+OE_WEAK
```

Comment:
This can be made a function pointer. That would make this PR likely work on Windows as well.

Reply:
updated.